website/project/views/drafts.py

```diff
@@ -0,0 +1,282 @@
+    """Submit for approvals and/or notifications
+
+    :return: serialized registration
+    :rtype: dict
+    """
+    # TODO(lyndsysimon): Is this view is called for all schemas?
+    approval = DraftRegistrationApproval(
+        initiated_by=auth.user,
+        end_date=None,
+    )
+    approval.save()
+    draft.approval = approval
+    draft.approval.ask(node.active_contributors())
+    draft.save()
+
+    ret = project_utils.serialize_node(node, auth)
+    ret['success'] = True
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def draft_before_register_page(auth, node, draft, *args, **kwargs):
+    """Allow the user to select an embargo period and confirm registration
+
+    :return: serialized Node + DraftRegistration
+    :rtype: dict
+    """
+    ret = serialize_node(node, auth, primary=True)
+
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+@http_error_if_disk_saving_mode
+def register_draft_registration(auth, node, draft, *args, **kwargs):
+    """Initiate a registration from a draft registration
+
+
+    :return: success message; url to registrations page
+    :rtype: dict
+    """
+
+    data = request.get_json()
+    register = draft.register(auth)
+    draft.save()
+
+    if data.get('registrationChoice', 'immediate') == 'embargo':
+        # Initiate embargo
+        embargo_end_date = parse_date(data['embargoEndDate'], ignoretz=True)
+        try:
+            register.embargo_registration(auth.user, embargo_end_date)
+        except ValidationValueError as err:
+            raise HTTPError(http.BAD_REQUEST, data=dict(message_long=err.message))
+    else:
+        try:
+            register.require_approval(auth.user)
+        except NodeStateError as err:
+            raise HTTPError(http.BAD_REQUEST, data=dict(message_long=err.message))
+
+    register.save()
+    push_status_message(language.AFTER_REGISTER_ARCHIVING,
+                        kind='info',
+                        trust=False)
+    return {
+        'status': 'initiated',
+        'urls': {
+            'registrations': node.web_url_for('node_registrations')
+        }
+    }, http.ACCEPTED
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registration(auth, node, draft, *args, **kwargs):
+    """Return a single draft registration
+
+    :return: serialized draft registration
+    :rtype: dict
+    """
+    return serialize_draft_registration(draft, auth), http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registrations(auth, node, *args, **kwargs):
+    """List draft registrations for a node
+
+    :return: serialized draft registrations
+    :rtype: dict
+    """
+
+    count = request.args.get('count', 100)
+    drafts = node.draft_registrations.find(
+        Q('registered_node', 'eq', None)
```

Comment:
Why do we use this particular filter for something being a draft registration? Under what conditions is a draft registration findable via backref (associated with a project), yet with a blank value of `registered_node` ?

(is this similar to the `DraftRegistration.is_draft_registration` flag? If so, I'd prefer there only be one obvious way of checking something)